zilliqa/src/consensus.rs

```diff
@@ -366,14 +484,7 @@ impl Consensus {
             Ok(()) => {}
             Err(e) => {
                 if let Some(e) = e.downcast_ref::<MissingBlockError>() {
-                    trace!(?e, "missing block");
-                    match e.0 {
-                        BlockRef::Hash(hash) => self.block_store.request_block(hash)?,
-                        BlockRef::View(view) => self.block_store.request_block_by_view(view)?,
-                    }
-
-                    self.add_block(block)?;
-
+                    info!(?e, "missing block when checking block proposal");
```

Comment:
IMO the network should send as few messages as possible. It should not request blocks etc when it detects a floating block, as this could cause network cascades (for example, when you start up, you will get a bunch of blocks from peers that will trigger this logic, but in reality you want to just patiently sync the chain)